ruby 2.7.0 overloaded rb_define_virtual_variable()

```
$ make                                                                                                                                                        
[  1%] Automatic MOC for target openshot                                                                                                                                                                           
[  1%] Built target openshot_autogen                                                                                                                                                                               
[ 68%] Built target openshot                                                                                                                                                                                       
[ 70%] Built target openshot-html-test                                                                                                                                                                             
[ 73%] Built target openshot-example                                                                                                                                                                               
[ 75%] Built target openshot-player                                                                                                                                                                                
[ 76%] Built target pyopenshot_swig_compilation                                                                                                                                                                    
[ 78%] Built target pyopenshot                                                                                                                                                                                     
[ 79%] Built target rbopenshot_swig_compilation                                                                                                                                                                    
[ 80%] Building CXX object src/bindings/ruby/CMakeFiles/rbopenshot.dir/CMakeFiles/rbopenshot.dir/openshotRUBY_wrap.cxx.o                                                                                           
/home/chiller/Desktop/git-extern/libopenshot/build/src/bindings/ruby/CMakeFiles/rbopenshot.dir/openshotRUBY_wrap.cxx: In function ‘void SWIG_RubyInitializeTrackings()’:                                           
/home/chiller/Desktop/git-extern/libopenshot/build/src/bindings/ruby/CMakeFiles/rbopenshot.dir/openshotRUBY_wrap.cxx:1263:84: error: call of overloaded ‘rb_define_virtual_variable(const char [21], VALUE (&)(...)
, NULL)’ is ambiguous                                                                                                                                                                                              
   rb_define_virtual_variable("SWIG_TRACKINGS_COUNT", swig_ruby_trackings_count, NULL);                                                                                                                            
                                                                                    ^                                                                                                                              
In file included from /home/chiller/.rbenv/versions/2.7.0/include/ruby-2.7.0/ruby/ruby.h:2863,                                                                                                                     
                 from /home/chiller/.rbenv/versions/2.7.0/include/ruby-2.7.0/ruby.h:33,                                                                                                                            
                 from /home/chiller/Desktop/git-extern/libopenshot/build/src/bindings/ruby/CMakeFiles/rbopenshot.dir/openshotRUBY_wrap.cxx:879:                                                                    
/home/chiller/.rbenv/versions/2.7.0/include/ruby-2.7.0/ruby/backward/cxxanyargs.hpp:59:1: note: candidate: ‘void ruby::backward::cxxanyargs::rb_define_virtual_variable(const char*, VALUE (*)(...), void (*)(...))
’                                                                                                                                                                                                                  
 rb_define_virtual_variable(const char *q, type *w, void_type *e)                                                                                                                                                  
 ^~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                        
/home/chiller/.rbenv/versions/2.7.0/include/ruby-2.7.0/ruby/backward/cxxanyargs.hpp:90:1: note: candidate: ‘void ruby::backward::cxxanyargs::rb_define_virtual_variable(const char*, VALUE (*)(...), void (*)(VALUE
, ID, VALUE*))’                                                                                                                                                                                                    
 rb_define_virtual_variable(const char *q, type *w, rb_gvar_setter_t *e)                                                                                                                                           
 ^~~~~~~~~~~~~~~~~~~~~~~~~~  
```

--- a/src/bindings/ruby/CMakeLists.txt
+++ b/src/bindings/ruby/CMakeLists.txt
@@ -41,6 +41,23 @@ if (NOT RUBY_FOUND)
   return()
 endif()
 
+### Ruby 2.7.0 made API changes that are incompatible with versions of
+### SWIG prior to 4.0.3
+option(SILENCE_RUBY_VERSION_WARNING
+  "Don't warn about possible SWIG incompatibilities with Ruby 2.7.0+" OFF)
+
+if (${RUBY_VERSION} VERSION_GREATER 2.6.9 AND ${SWIG_VERSION} VERSION_LESS 4.0.3)
+  if (NOT ${SILENCE_RUBY_VERSION_WARNING})
+    message(WARNING "
+Ruby 2.7.0+ detected, building the libopenshot Ruby API bindings \
+requires a pre-release version of SWIG 4.0.3 with this commit: \
+https://github.com/swig/swig/commit/5542cc228ad10bdc5c91107afb77c808c43bf2a4")
+    message(STATUS "
+To disable this warning, add -DSILENCE_RUBY_VERSION_WARNING:BOOL=1 to the cmake \
+command line, or enable the option in the CMake GUI.")
+  endif()
+endif()
+
 ### Include the Ruby header files
 include_directories(${RUBY_INCLUDE_DIRS})
 

